Src/Extern/HYPRE/AMReX_HypreSolver.H

```diff
@@ -0,0 +1,665 @@
+#ifndef AMREX_HYPRE_SOLVER_H_
+#define AMREX_HYPRE_SOLVER_H_
+
+#include <AMReX_Geometry.H>
+#include <AMReX_iMultiFab.H>
+#include <AMReX_HypreIJIface.H>
+#include <AMReX_BLProfiler.H>
+
+#include "HYPRE.h"
+#include "_hypre_utilities.h"
+
+#include <string>
+
+namespace amrex
+{
+
+template <int MSS>
+class HypreSolver
+{
+public:
+
+    template <class Marker, class Filler>
+    HypreSolver (Vector<IndexType>   const& a_index_type,
+                 IntVect             const& a_nghost,
+                 Geometry            const& a_geom,
+                 BoxArray            const& a_grids,
+                 DistributionMapping const& a_dmap,
+                 Marker                  && a_marker,
+                 Filler                  && a_filler,
+                 int                        a_verbose = 0,
+                 std::string         const& a_options_namespace = "hypre");
+
+    template <class MF, std::enable_if_t<IsFabArray<MF>::value &&
+                                         std::is_same_v<typename MF::value_type,
+                                                        HYPRE_Real>, int> = 0>
+    void solve (Vector<MF      *> const& a_soln,
+                Vector<MF const*> const& a_rhs,
+                HYPRE_Real rel_tol, HYPRE_Real abs_tol, int max_iter);
+
+    int getNumIters () const { return m_hypre_ij->getNumIters(); }
+
+    HYPRE_Real getFinalResidualNorm () const {
+        return m_hypre_ij->getFinalResidualNorm();
+    }
+
+    HYPRE_IJMatrix getA () const { return m_hypre_ij->A(); }
+    HYPRE_IJVector getb () const { return m_hypre_ij->b(); }
+    HYPRE_IJVector getx () const { return m_hypre_ij->x(); }
+
+public: // for cuda
+
+    template <class Marker>
+#ifdef AMREX_USE_CUDA
```

Comment:
```suggestion
#ifdef AMREX_USE_GPU
```
To be consistent with more broad ifdef below

Reply:
The macro is a workaround for *nvcc* compiler.